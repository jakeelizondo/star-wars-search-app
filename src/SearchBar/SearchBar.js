import React from 'react';
import AppContext from '../AppContext';

class SearchBar extends React.Component {
  state = { searchTerm: '', searchCategory: 'none', touched: false };
  static contextType = AppContext;

  handleSubmit = (e) => {
    e.preventDefault();
    this.context.getSearchResults(
      this.state.searchTerm,
      this.state.searchCategory
    );
  };

  validateEntry = () => {
    if (this.state.searchTerm === '' || this.state.searchCategory === 'none') {
      return true;
    }
  };

  render() {
    return (
      <form onSubmit={(e) => this.handleSubmit(e)}>
        <h2>Search the API</h2>
        <div>
          <label htmlFor="category">Search for: </label>
          <select
            id="categories"
            onChange={(event) =>
              this.setState({ searchCategory: event.target.value })
            }
          >
            <option value="none">Select a category...</option>
            <option>films</option>
            <option>people</option>
            <option>planets</option>
            <option>species</option>
            <option>starships</option>
            <option>vehicles</option>
          </select>
          {this.state.searchCategory === 'none' && this.state.touched && (
            <p style={{ color: 'red' }}>Please select a category.</p>
          )}
        </div>
        <br />
        <div>
          <label htmlFor="search">that include the term: </label>
          <input
            type="text"
            id="search"
            onChange={(event) =>
              this.setState({ searchTerm: event.target.value, touched: true })
            }
          ></input>
          {this.state.touched && this.state.searchTerm === '' && (
            <p style={{ color: 'red' }}>This field cannot be left blank</p>
          )}
          <button type="submit" disabled={this.validateEntry()}>
            Search
          </button>
        </div>
      </form>
    );
  }
}

export default SearchBar;
