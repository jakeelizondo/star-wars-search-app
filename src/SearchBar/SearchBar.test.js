import React from 'react';
import ReactDOM from 'react-dom';
import SearchBar from './SearchBar';
import renderer from 'react-test-renderer';

describe('SearchBar', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SearchBar />, div);

    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer.create(<SearchBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
