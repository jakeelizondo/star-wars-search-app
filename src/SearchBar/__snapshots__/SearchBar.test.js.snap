// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders the UI as expected 1`] = `
<form
  onSubmit={[Function]}
>
  <h2>
    Search the API
  </h2>
  <div>
    <label
      htmlFor="category"
    >
      Search for: 
    </label>
    <select
      id="categories"
      onChange={[Function]}
    >
      <option
        value="none"
      >
        Select a category...
      </option>
      <option>
        films
      </option>
      <option>
        people
      </option>
      <option>
        planets
      </option>
      <option>
        species
      </option>
      <option>
        starships
      </option>
      <option>
        vehicles
      </option>
    </select>
  </div>
  <br />
  <div>
    <label
      htmlFor="search"
    >
      that include the term: 
    </label>
    <input
      id="search"
      onChange={[Function]}
      type="text"
    />
    <button
      disabled={true}
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
